src/vscode/commands/Repl.ts

```diff
@@ -63,6 +67,35 @@ export async function inlineEval(state: ExtensionState) {
     })
 }
 
+export async function startReplAndAttach(state: ExtensionState, ctx: vscode.ExtensionContext) {
+    const cmd = vscode.workspace.getConfiguration('alive').swank.startCommand;
+    const outputAndAttach = (data: string) => {
+        outputChannel.appendLine(data)
+
+        if (data.includes("Swank started at port")) {
+            attachRepl(state, ctx)
+        }
+    }
+    const spawn = () => {
+        child = childProcess.spawn(cmd[0], cmd.slice(1))
+        child.stdout.setEncoding('utf8');
+        child.stdout.on('data', data => outputAndAttach(data));
+
+        child.stderr.setEncoding('utf8');
+        child.stderr.on('data', data => outputAndAttach(data));
+    }
+    if (child) {
+        child.kill()
+        vscode.window.showInformationMessage("Waiting for Swank to die...")
+        // For some reason 'exit' event wasn't getting called every time so I'm using timeout.
```

Comment:
Is there a way to reproduce exit not being fired?
The right way to do this is probably to have something like a killChild() call that creates a promise that resolves on the exit event.
Then call await killChild() before spawn().
The callback to the original child exit handler would need to be kept in a variable so that it can be turned off before explicitly killing the child.

Reply:
Ok so I've found the reason why 'exit' wasn't getting called, my mistake, I was overriding already existing handler.
I've removed `spawn` function as it's not needed anymore and added 'error' listener.
Adding 'exit' listener isn't necessary imho as simply setting child to null just after killing it seems to be enough.

Reply:
It needs to wait for exit. Not waiting creates a race condition where the new process could try to get the port before the old one has released it.